src/custom-element.js

```diff
@@ -602,26 +602,31 @@ function createBaseCustomElementClass(win) {
           return this.whenLoaded();
         }
 
+        // Very ugly! The "built" signal must be resolved from the Resource
+        // and not the element itself because the Resource has not correctly
+        // set its state for the downstream to process it correctly.
```

Comment:
what is downstream? would it be cleanest for custom-element to force `built` signal on Resource?

Reply:
Unfortunately, no. The problem is not only a built signal, but also all this code: https://github.com/ampproject/amphtml/blob/c3e45ef8633190ac1fc6a6f8cc6b84512eaad2f0/src/service/resource.js#L346-L362

What we could do is to refactor that code to be called by the `CustomElement` itself in the same microtask as its own [build completion code](https://github.com/ampproject/amphtml/blob/c3e45ef8633190ac1fc6a6f8cc6b84512eaad2f0/src/custom-element.js#L509-L513).

Reply:
This would actually be quite easy to achieve. But I still wonder if doing this would take away time from maybe porting a few more elements to V1 to address most of the issues. Let me know what you think.

Reply:
Is it that both Resource and CustomElement both have their own `isBuilt` state and we need to dedupe them or are there other issues? I'm in favor of deduping as much as possible between Resource and CustomElement (sizes, states, etc)

Reply:
Yes. Big area of concern. V1 elements so far simply do not use `Resource` interface at all (i.e. they are already deduped). We can definitely dedupe V0 elements as well area-by-area.